### ISSUE TITLE
Volume mount timeout

### ISSUE THREAD

**Issue by hainabc** [purestorage/helm-charts#57]
We try to use our Flash Array to provide persistent storage for our Openshift cluster but I keep getting

Unable to mount volumes for pod "docker-registry-7-2wr7g_default(d19d07e2-1e88-11e9-8fbd-0050569a062c)": timeout expired waiting for volumes to attach or mount for pod

The volume was created just fine on the array so, the plugin can create the volume but it wasn't be able to mount the volume to the pod.
We did change sanType in the value.yaml to FC. 
The nodes are also boot from the same array. So, they can see the Array via FC.
Not sure what I miss here.

**Comment by dinathom** [comment]
@hainabc  we need the syslog from the time of this incident to investigate. Please can you provide this. Also I would highly encourage you to file a  support ticket so we can track the issue and progress. Thanks!

**Comment by hainabc** [comment]
Here is the /var/log/messages starting from where I start the deployment

Jan 22 21:59:27 dc01osnode01 atomic-openshift-node: I0122 21:59:27.236287   35042 kubelet.go:1820] SyncLoop (ADD, "api"): "docker-registry-9-deploy_default(e55bd5f4-1eba-11e9-8d42-0050569a2196)"
Jan 22 21:59:27 dc01osnode01 systemd: Created slice libcontainer container kubepods-besteffort-pode55bd5f4_1eba_11e9_8d42_0050569a2196.slice.
Jan 22 21:59:27 dc01osnode01 atomic-openshift-node: I0122 21:59:27.323846   35042 reconciler.go:207] operationExecutor.VerifyControllerAttachedVolume started for volume "deployer-token-tzfs6" (UniqueName: "kubernetes.io/secret/e55bd5f4-1eba-11e9-8d42-0050569a2196-deployer-token-tzfs6") pod "docker-registry-9-deploy" (UID: "e55bd5f4-1eba-11e9-8d42-0050569a2196")
Jan 22 21:59:27 dc01osnode01 atomic-openshift-node: I0122 21:59:27.424322   35042 reconciler.go:252] operationExecutor.MountVolume started for volume "deployer-token-tzfs6" (UniqueName: "kubernetes.io/secret/e55bd5f4-1eba-11e9-8d42-0050569a2196-deployer-token-tzfs6") pod "docker-registry-9-deploy" (UID: "e55bd5f4-1eba-11e9-8d42-0050569a2196")
Jan 22 21:59:27 dc01osnode01 systemd: Started Kubernetes transient mount for /var/lib/origin/openshift.local.volumes/pods/e55bd5f4-1eba-11e9-8d42-0050569a2196/volumes/kubernetes.io~secret/deployer-token-tzfs6.
Jan 22 21:59:27 dc01osnode01 atomic-openshift-node: I0122 21:59:27.441975   35042 operation_generator.go:558] MountVolume.SetUp succeeded for volume "deployer-token-tzfs6" (UniqueName: "kubernetes.io/secret/e55bd5f4-1eba-11e9-8d42-0050569a2196-deployer-token-tzfs6") pod "docker-registry-9-deploy" (UID: "e55bd5f4-1eba-11e9-8d42-0050569a2196")
Jan 22 21:59:27 dc01osnode01 atomic-openshift-node: I0122 21:59:27.569723   35042 kuberuntime_manager.go:385] No sandbox for pod "docker-registry-9-deploy_default(e55bd5f4-1eba-11e9-8d42-0050569a2196)" can be found. Need to start a new one
Jan 22 21:59:27 dc01osnode01 systemd: Started libcontainer container 2dd17794a1ba160224025da2b435d733f0411bce55d54f2afe9d2dc66037b04c.
Jan 22 21:59:27 dc01osnode01 kernel: SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
Jan 22 21:59:27 dc01osnode01 kernel: IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Jan 22 21:59:27 dc01osnode01 kernel: IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Jan 22 21:59:27 dc01osnode01 NetworkManager[29808]: <info>  [1548212367.8036] device (vethb44f7ece): carrier: link connected
Jan 22 21:59:27 dc01osnode01 NetworkManager[29808]: <info>  [1548212367.8040] manager: (vethb44f7ece): new Veth device (/org/freedesktop/NetworkManager/Devices/28)
Jan 22 21:59:27 dc01osnode01 NetworkManager[29808]: <info>  [1548212367.8231] device (vethb44f7ece): enslaved to non-master-type device ovs-system; ignoring
Jan 22 21:59:27 dc01osnode01 kernel: device vethb44f7ece entered promiscuous mode
Jan 22 21:59:28 dc01osnode01 atomic-openshift-node: I0122 21:59:28.531964   35042 kubelet.go:1865] SyncLoop (PLEG): "docker-registry-9-deploy_default(e55bd5f4-1eba-11e9-8d42-0050569a2196)", event: &pleg.PodLifecycleEvent{ID:"e55bd5f4-1eba-11e9-8d42-0050569a2196", Type:"ContainerStarted", Data:"2dd17794a1ba160224025da2b435d733f0411bce55d54f2afe9d2dc66037b04c"}
Jan 22 21:59:29 dc01osnode01 systemd: Started libcontainer container 2d67f3b2a9ce0f432ae2b0e195db0842cfce47acfa5e6220b599333e759d3c74.
Jan 22 21:59:29 dc01osnode01 kernel: SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
Jan 22 21:59:29 dc01osnode01 atomic-openshift-node: I0122 21:59:29.558882   35042 kubelet.go:1865] SyncLoop (PLEG): "docker-registry-9-deploy_default(e55bd5f4-1eba-11e9-8d42-0050569a2196)", event: &pleg.PodLifecycleEvent{ID:"e55bd5f4-1eba-11e9-8d42-0050569a2196", Type:"ContainerStarted", Data:"2d67f3b2a9ce0f432ae2b0e195db0842cfce47acfa5e6220b599333e759d3c74"}
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: I0122 21:59:30.396301   35042 kubelet.go:1820] SyncLoop (ADD, "api"): "docker-registry-9-94w9v_default(e73ce0f3-1eba-11e9-8fbd-0050569a062c)"
Jan 22 21:59:30 dc01osnode01 systemd: Created slice libcontainer container kubepods-burstable-pode73ce0f3_1eba_11e9_8fbd_0050569a062c.slice.
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: E0122 21:59:30.411055   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: I0122 21:59:30.432899   35042 reconciler.go:207] operationExecutor.VerifyControllerAttachedVolume started for volume "registry-certificates" (UniqueName
: "kubernetes.io/secret/e73ce0f3-1eba-11e9-8fbd-0050569a062c-registry-certificates") pod "docker-registry-9-94w9v" (UID: "e73ce0f3-1eba-11e9-8fbd-0050569a062c")
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: I0122 21:59:30.432953   35042 reconciler.go:207] operationExecutor.VerifyControllerAttachedVolume started for volume "registry-token-pnmvg" (UniqueName:
 "kubernetes.io/secret/e73ce0f3-1eba-11e9-8fbd-0050569a062c-registry-token-pnmvg") pod "docker-registry-9-94w9v" (UID: "e73ce0f3-1eba-11e9-8fbd-0050569a062c")
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: E0122 21:59:30.522511   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: I0122 21:59:30.533364   35042 reconciler.go:252] operationExecutor.MountVolume started for volume "registry-certificates" (UniqueName: "kubernetes.io/secret/e73ce0f3-1eba-11e9-8fbd-0050569a062c-registry-certificates") pod "docker-registry-9-94w9v" (UID: "e73ce0f3-1eba-11e9-8fbd-0050569a062c")
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: I0122 21:59:30.533473   35042 reconciler.go:252] operationExecutor.MountVolume started for volume "registry-token-pnmvg" (UniqueName: "kubernetes.io/secret/e73ce0f3-1eba-11e9-8fbd-0050569a062c-registry-token-pnmvg") pod "docker-registry-9-94w9v" (UID: "e73ce0f3-1eba-11e9-8fbd-0050569a062c")
Jan 22 21:59:30 dc01osnode01 systemd: Started Kubernetes transient mount for /var/lib/origin/openshift.local.volumes/pods/e73ce0f3-1eba-11e9-8fbd-0050569a062c/volumes/kubernetes.io~secret/registry-certificates.
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: I0122 21:59:30.550049   35042 operation_generator.go:558] MountVolume.SetUp succeeded for volume "registry-certificates" (UniqueName: "kubernetes.io/secret/e73ce0f3-1eba-11e9-8fbd-0050569a062c-registry-certificates") pod "docker-registry-9-94w9v" (UID: "e73ce0f3-1eba-11e9-8fbd-0050569a062c")
Jan 22 21:59:30 dc01osnode01 systemd: Started Kubernetes transient mount for /var/lib/origin/openshift.local.volumes/pods/e73ce0f3-1eba-11e9-8fbd-0050569a062c/volumes/kubernetes.io~secret/registry-token-pnmvg.
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: I0122 21:59:30.557363   35042 operation_generator.go:558] MountVolume.SetUp succeeded for volume "registry-token-pnmvg" (UniqueName: "kubernetes.io/secret/e73ce0f3-1eba-11e9-8fbd-0050569a062c-registry-token-pnmvg") pod "docker-registry-9-94w9v" (UID: "e73ce0f3-1eba-11e9-8fbd-0050569a062c")
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: E0122 21:59:30.632595   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:30 dc01osnode01 atomic-openshift-node: E0122 21:59:30.803194   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:31 dc01osnode01 atomic-openshift-node: E0122 21:59:31.203389   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:31 dc01osnode01 atomic-openshift-node: E0122 21:59:31.597557   35042 summary.go:102] Failed to get system container stats for "/system.slice/atomic-openshift-node.service": failed to get cgroup stats for "/system.slice/atomic-openshift-node.service": failed to get container info for "/system.slice/atomic-openshift-node.service": unknown container "/system.slice/atomic-openshift-node.service"
Jan 22 21:59:31 dc01osnode01 atomic-openshift-node: E0122 21:59:31.602548   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:32 dc01osnode01 atomic-openshift-node: E0122 21:59:32.001247   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:32 dc01osnode01 atomic-openshift-node: E0122 21:59:32.403197   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:32 dc01osnode01 atomic-openshift-node: E0122 21:59:32.802894   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:33 dc01osnode01 atomic-openshift-node: E0122 21:59:33.202952   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
Jan 22 21:59:33 dc01osnode01 atomic-openshift-node: E0122 21:59:33.602900   35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched

After that, the log just spamming with the previous message Failed to add volume.
So, I think the provisioner works, but the flexvolume driver is not.
I manually connect the volume to the host on the Array and rescan scsi and multipath can pick up the volume just fine, so connectivity between the host and the array is ok.
Where do you want me to open a support ticket? Pure1?

**Comment by taherv** [comment]
```
Jan 22 21:59:33 dc01osnode01 atomic-openshift-node: E0122 21:59:33.602900 35042 desired_state_of_world_populator.go:311] Failed to add volume "registry-storage" (specName: "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196") for pod "e73ce0f3-1eba-11e9-8fbd-0050569a062c" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-4d98d0ce-1e99-11e9-8d42-0050569a2196" err=no volume plugin matched
```
The no volume plugin matched message is the critical error here.
It means that kubelet is not finding Pure's flexvol driver binary. Kubelet looks in a well known location, and Pure's PSO install usually copies its flexvol driver binaries in the right location.
To see where PSO is dropping binaries, look at the volumes mounted by the pure-flex pod. There should be one per node. Looking at any one is okay. Look at the mount source for /kubelet-plugins in the output of `kubectl describe pod <pure-flex-pod`.
To see where kubelet might be searching for our flexvol binary, you will have to login to the node and look at kubelet's command line parameters.

**Comment by hainabc** [comment]
Here is the describe pod result

Name:               pure-flex-f4s7d
Namespace:          pure-storage
Priority:           0
PriorityClassName:  <none>
Node:               dc01osnode01.abcfinancial.net/172.20.109.12
Start Time:         Tue, 22 Jan 2019 14:59:00 -0600
Labels:             app=pure-flex
                    chart=pure-k8s-plugin
                    controller-revision-hash=2717509372
                    generator=helm
                    pod-template-generation=3
                    release=pure-storage-driver
Annotations:        openshift.io/scc=privileged
Status:             Running
IP:                 10.129.4.17
Controlled By:      DaemonSet/pure-flex
Containers:
  pure-flex:
    Container ID:  docker://a255d91a83c7c5781c92750ae5ce721f7d0c6d0e0fc9042cedd76f973388e109
    Image:         purestorage/k8s:2.2.1
    Image ID:      docker-pullable://docker.io/purestorage/k8s@sha256:c98e4cb5e5223221b35c36cd43701603d4942b48af7ffe93ba7d380e9980e5c0
    Port:          <none>
    Host Port:     <none>
    Command:
      /pure-flex-daemon
      --kubeletBaseDir=/kubelet-plugins
      --daemon
      install
    State:          Running
      Started:      Tue, 22 Jan 2019 14:59:02 -0600
    Ready:          True
    Restart Count:  0
    Environment:
      PURE_DISCOVERY_CONF:                   /etc/pure/pure.json
      PURE_FLASHARRAY_SAN_TYPE:              FC
      PURE_K8S_NAMESPACE:                    pure-storage
      PURE_DEFAULT_BLOCK_FS_TYPE:            xfs
      PURE_DEFAULT_BLOCK_FS_OPT:             -q
      PURE_DEFAULT_BLOCK_MNT_OPT:
      PURE_PREEMPT_RWO_ATTACHMENTS_DEFAULT:  true
    Mounts:
      /etc/pure from config (rw)
      /kubelet-plugins from kubelet-plugins (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from pure-token-dbm9b (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      pure-provisioner-config
    Optional:  false
  kubelet-plugins:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/origin/node/kubelet-plugins/volume/exec
    HostPathType:
  pure-token-dbm9b:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  pure-token-dbm9b
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  node-role.kubernetes.io/compute=true
Tolerations:     node.kubernetes.io/disk-pressure:NoSchedule
                 node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/unreachable:NoExecute
Events:          <none>


So, it looks like the path is /etc/origin/node/kubelet-plugins/volume/exec
I checked this path 

[root@dc01osnode01 ~]# cd /etc/origin/node/kubelet-plugins/volume/exec/
[root@dc01osnode01 exec]# ls -la
total 0
drwxr-xr-x. 3 root root 23 Jan 22 15:59 .
drwxr-xr-x. 3 root root 18 Jan 22 13:31 ..
drwxr-xr-x. 2 root root 54 Jan 22 17:56 pure~flex

[root@dc01osnode01 exec]# cd pure~flex/
[root@dc01osnode01 pure~flex]# ls -la
total 11496
drwxr-xr-x. 2 root root       54 Jan 22 17:56 .
drwxr-xr-x. 3 root root       23 Jan 22 15:59 ..
-rwxr-xr-x. 1 root root      420 Jan 22 15:59 flex
-rwxr-xr-x. 1 root root 11762840 Jan 22 15:59 .pure-flex
-rw-r--r--. 1 root root      139 Jan 22 17:56 .pure.json

The .pure-flex is the binary. So, it's there.

**Comment by taherv** [comment]
Yep, the path is correct. And it is the default path in the helm chart from here :
https://github.com/purestorage/helm-charts/blob/348474c12fb415d80d7e4035a125bc22ee8b26ad/pure-k8s-plugin/values.yaml#L49

The next thing to check, does kubelet look in this path ?
Also, what version of Kubernetes/Openshift is this ? (You can run kubectl version, it should print client and server versions, we are interested in the Server version here). I bring this up, because prior to Kubernetes v1.9, we had to restart kubelet after installing our flexvol driver. Versions 1.9 and above detect binaries being dropped dynamically, older versions only inspect available drivers at kubelet startup. I suspect this might not be the problem, no harm in checking.

**Comment by hainabc** [comment]
I found the problem. I am current running Openshift 3.11, in the document, Openshift is looking for flexvolume driver at /usr/libexec/kubernetes/kubelet-plugins/volume/exec/ instead.
So, I created a pure~flex folder and copy everything from /etc/origin/node/kubelet-plugins/volume/exec to it and the pod can now mount the volume.
It looks like your base install path need to change, at least for OCP 3.11

**Comment by taherv** [comment]
Glad you could adjust the paths. The values can be overridden at install time, so I suggest you adjust the path there. This manual copy might not stick if you reimage your node, plus you have to do this manually for all nodes. That's what the pure-flex daemon is for :)

I'll keep this bug open to implement the fix for 3.11 as you suggested, thanks for updating the bug and following up on my suggestions .....

**Comment by hainabc** [comment]
Which looks like OCP is matching with up-steam Kubernetes. Heh, now you don't have to split in 2 anymore

**Comment by hainabc** [comment]
Also, looks like for Openshift, the flexvolume driver only create pod to compute nodes only

https://github.com/purestorage/helm-charts/blob/348474c12fb415d80d7e4035a125bc22ee8b26ad/pure-k8s-plugin/templates/pure-flex-daemon.yaml#L20-L23

This probably is not ideal since there are things schedule in the infrastructure nodes and master nodes need storage as well (Openshift internal docker registry is one).

I suggest to remove those lines. The default cluster wide node selector for Openshift is compute nodes already, so without doing anything, removing those lines would still give you the same behavior like it is now.

If the user want to schedule the flexvolume pod to more than just compute (I would, in my case) they just need to add 

`metadata:
  annotations:
    openshift.io/description: ""`

to the project/namespace where they deploy pure driver and the flexvolume will schedule to all nodes, or they can do their own label.

**Comment by dinathom** [comment]
Regarding the location of the flexvolume binary. It is related to whether you are running RHEL Server or RHEL Atomic Host. @hainabc are you running RHEL Server? 

As pointed out here: https://docs.openshift.com/container-platform/3.11/install_config/persistent_storage/persistent_storage_flex_volume.html

Quoting : "In OpenShift Container Platform 3.11, since controller-manager runs as a static pod, the FlexVolume binary file that performs the attach and detach operations must be a self-contained executable file with no external dependencies.

On Atomic hosts, the default location of the FlexVolume plug-in directory is /etc/origin/kubelet-plugins/. You must place the FlexVolume executable file in the /etc/origin/kubelet-plugins/volume/exec/<vendor>~<driver>/<driver> directory on all master and nodes in the cluster."

**Comment by hainabc** [comment]
I am running RHEL Server. That make senses now.

**Comment by dinathom** [comment]
Addressed by commit 9871d644fc9660e671870fbe5cf6e0320dd526d1

**Comment by dburkland** [comment]
For what it's worth on RKE 1.14/15/16 you need to set the FlexPath setting equal to "/var/lib/kubelet/volumeplugins". Once that is set correctly the PSO works great.
